src/app/backend/deploy.go

```diff
@@ -231,3 +246,65 @@ func getLabelsMap(labels []Label) map[string]string {
 
 	return result
 }
+
+// Deploys an app based on the given yaml or json file.
+func DeployAppFromFile(spec *AppDeploymentFromFileSpec, client *client.Client) error {
+	cmd := create.NewCmdCreate(cmdutil.NewFactory(nil), os.Stdout)
+	return RunCreate(cmdutil.NewFactory(nil), cmd, os.Stdout, spec)
+}
+
+// Validates and runs the given command
+func RunCreate(f *cmdutil.Factory, cmd *cobra.Command, out io.Writer, spec *AppDeploymentFromFileSpec) error {
+	schema, err := f.Validator(cmdutil.GetFlagBool(cmd, "validate"), cmdutil.GetFlagString(cmd, "schema-cache-dir"))
```

Comment:
schema-cache-dir is created in NewCmdCreate method by the constants defined in loader.go. If I create copy of these constants we can't get any update of these constants in case somebody changes them. For validate I can use a constant which is set to true since it is just a boolean value.